patches/net/minecraft/world/item/ArmorItem.java.patch

```diff
@@ -0,0 +1,10 @@
+--- a/net/minecraft/world/item/ArmorItem.java
++++ b/net/minecraft/world/item/ArmorItem.java
+@@ -49,6 +_,7 @@
+         } else {
+             LivingEntity livingentity = list.get(0);
+             EquipmentSlot equipmentslot = livingentity.getEquipmentSlotForItem(p_40400_);
++            if (!p_40400_.canEquip(equipmentslot, livingentity)) return false;
```

Comment:
Needs a comment declaring the purpose:
`// Neo: Respect IItemExtension#canEquip in dispenseArmor`

Reply:
fixed